dizoo/classic_control/pendulum/envs/pendulum_env.py

```diff
@@ -58,12 +61,17 @@ def seed(self, seed: int, dynamic_seed: bool = True) -> None:
 
     def step(self, action: np.ndarray) -> BaseEnvTimestep:
         assert isinstance(action, np.ndarray), type(action)
+        # if require discrete env, convert actions to [-2 ~ 2] float actions
+        if not self._is_continue:
+            action=(action-(self._discrete_action_num-1)/2)/((self._discrete_action_num-1)/4)
+        # why scale? into [-2, 2]
         if self._act_scale:
             action = affine_transform(action, min_val=self._env.action_space.low, max_val=self._env.action_space.high)
         obs, rew, done, info = self._env.step(action)
         self._final_eval_reward += rew
         obs = to_ndarray(obs).astype(np.float32)
-        rew = to_ndarray([rew]).astype(np.float32)  # wrapped to be transfered to a array with shape (1,)
+        # rew = to_ndarray([rew]).astype(np.float32)  # wrapped to be transfered to a array with shape (1,)
+        rew = np.array([rew], dtype=np.float32)
```

Comment:
why use this implementation, it has no difference with original version

Reply:
Because I ran into such a problem when testing: `AttributeError: 'list' object has no attribute 'shape'`. So I'm guessing if it's the problem here, and later I find out that it's the problem with the parameters passed in earlier.